src/main/java/org/commcare/suite/model/Entry.java

```diff
@@ -157,14 +158,22 @@ public String getDisplayText() {
     }
 
     @Override
-    public Single<String> getTextForBadge(final EvaluationContext ec) {
+    public Single<String> getTextForBadge(EvaluationContext ec) {
+        final EvaluationContext[] abandonableContext =
```

Comment:
What is the reason for making this an array of ECs?

Reply:
This lets the the "dispose" action remove the reference to the abandonable context immediately. Since the Callable object is going into this whole event handling subsystem, I wanted to be super careful about leaving around references to the lifecycle objects and risking them being memory leaks.